wp_api/src/request.rs

```diff
@@ -753,6 +756,62 @@ pub fn is_valid_json(value: impl AsRef<str>) -> bool {
     serde_json::from_str::<serde_json::Value>(value.as_ref()).is_ok()
 }
 
+#[uniffi::export]
+pub async fn is_valid_authentication(
+    request_executor: Arc<dyn RequestExecutor>,
+    api_root_url: Arc<ParsedUrl>,
+    authentication_provider: Arc<WpAuthenticationProvider>,
+) -> IsValidAuthenticationResult {
+    let request = ApplicationPasswordsRequestBuilder::new(api_root_url, authentication_provider)
+        .retrieve_current_with_edit_context()
+        .into();
+    match request_executor.execute(request).await {
+        Ok(response) => {
+            let parsed_res: Result<
+                ApplicationPasswordsRequestRetrieveCurrentWithEditContextResponse,
+                WpApiError,
+            > = response.parse();
```

Comment:
I _think_ you can reuse the `ApplicationPasswordsRequestExecutor` to do this parsing for you?

Reply:
That'd potentially introduce a recursion here, because `ApplicationPasswordsRequestExecutor` will also potentially trigger the method. We also may introduce other behavior, middleware etc that impact what `ApplicationPasswordsRequestExecutor` does. Doing it this way avoids all of that, because we only build the request and manually execute it.

Reply:
This reminds me that we don't need to send an introspect API call if the original response is `WpErrorCode::Unauthorized`.

Reply:
This should address that: 99990ac904542c14f7c1af03d033c712d1eebf20